### PR TITLE
Small improvements in ci-entrypoint script 

### PR DESCRIPTION
**Pull request by CecileRobertMichon**
**What type of PR is this?**

**What this PR does / why we need it**: Broke out unrelated changes from #1330. Add a random region selection to avoid quota issues, refactored ACR login into a script and made sure REGISTRY is set, updated ACR cleanup tasks (see https://github.com/kubernetes/test-infra/pull/21933), and fixed a typo in cert-manager version.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: hack/ensure-acr-cleanup-schedule.sh [hunk 1]
```diff
@@ -18,10 +18,10 @@ set -o errexit
...
-e2e_registry="ci-e2e/cluster-api-azure-controller-amd64"
-az acr task create --name midnight_capz_ci_e2e_purge --cmd "acr purge --filter ${e2e_registry}:.* --ago 1d --untagged" \
-  --schedule "0 0 * * *" --registry capzci --context /dev/null
-
 registry="cluster-api-azure-controller-amd64"
 az acr task create --name midnight_capz_purge --cmd "acr purge --filter ${registry}:.* --ago 1d --untagged" \
   --schedule "0 0 * * *" --registry capzci --context /dev/null
+
+k8s_ci_registry="prow"
+az acr task create --name midnight_capz_purge --cmd "acr purge --filter ${k8s_ci_registry}:.* --ago 1d --untagged" \
```
Comment (chewong) [inline hack/ensure-acr-cleanup-schedule.sh:26]: I think this will only purge `prow` but not `prow/.*:.*`

Comment (CecileRobertMichon) [reply hack/ensure-acr-cleanup-schedule.sh]: testing

Comment (CecileRobertMichon) [reply hack/ensure-acr-cleanup-schedule.sh]: yeah you're right this isn't going to work... the "repository" gets created as `prow/cluster-api-azure-controller-amd64` and running this cleanup task fails with:

`Deleting manifests for repository: prow
prow repository not found`

Looks like we can do regex on the image tags but not on the image name

/hold

Comment (CecileRobertMichon) [reply hack/ensure-acr-cleanup-schedule.sh]:
```
Deleting tags for repository: prow/*
prow/* repository not found
Deleting manifests for repository: prow/*
prow/* repository not found
```

confirmed

**Review by devigned — APPROVED** [review]
Like what you've done with the `ensure-acr-cleanup-schedule.sh`. What do you think about an issue to add that script to a periodic job to ensure the job says up-to-date?

/lgtm
/approve

**Pull request merged**